dask/array/core.py

```diff
@@ -1177,7 +1177,10 @@ def __dask_postcompute__(self):
         return finalize, ()
 
     def __dask_postpersist__(self):
-        return Array, (self.name, self.chunks, self.dtype, self._meta)
+        return self._rebuild, ()
+
+    def _rebuild(self, dsk, name=None):
+        return Array(dsk, name or self.name, self.chunks, self.dtype, self._meta)
```

Comment:
This has the side effect that, if anybody were to pickle the output of ``__dask_postpersist__``, he would needlessly pickle the old dask graph too. However I cannot think of any real-life use case where this could happen?